py/server/deephaven/time.py

```diff
@@ -155,1885 +135,485 @@ def time_zone_alias_rm(alias: str) -> bool:
 
 # endregion
 
-# region Conversions: Time Units
-
+#TODO: Review all function names: to_j_<xyz>?  to_db_<xyz>?  to_dh_<xyz>?  to_<xyz>? as_j_<xyz>?  as_db_<xyz>?  as_dh_<xyz>?
+#TODO: Should the doc strings refer to "Deephaven" or "Java" types?
 
-def micros_to_nanos(micros: int) -> int:
-    """ Converts microseconds to nanoseconds.
-
-    Args:
-        micros (int): Microseconds to convert.
 
-    Returns:
-        NULL_LONG if the input is NULL_LONG; otherwise the input microseconds converted to nanoseconds.
+# region Conversions: Python To Java
 
-    Raises:
-        DHError
+def to_j_time_zone(tz: Union[None, str, datetime.tzinfo, datetime.datetime]) -> Optional[TimeZone]:
     """
-    try:
-        return _JDateTimeUtils.microsToNanos(micros)
-    except Exception as e:
-        raise DHError(e) from e
-
-
-def millis_to_nanos(millis: int) -> int:
-    """ Converts milliseconds to nanoseconds.
+    Converts a Python string, tzinfo, or datetime to a Deephaven time zone.
 
     Args:
-        millis (int): Milliseconds to convert.
+        tz (Union[None, str, datetime.tzinfo, datetime.datetime]): A Python time, time zone, or time zone string.
+            If None is provided, the Deephaven system default time zone is returned.
+            If a string is provided, it is parsed as a time zone name.
 
     Returns:
-        NULL_LONG if the input is NULL_LONG; otherwise the input milliseconds converted to nanoseconds.
+        TimeZone
 
     Raises:
         DHError
     """
     try:
-        return _JDateTimeUtils.millisToNanos(millis)
+        if tz is None:
+            return _JDateTimeUtils.timeZone()
+            #TODO: return None or system default?
+        elif isinstance(tz, str):
+            return _JDateTimeUtils.parseTimeZone(tz)
+        elif isinstance(tz, datetime.tzinfo):
+            return _JDateTimeUtils.parseTimeZone(str(tz))
+        elif isinstance(tz, datetime.datetime):
+            if not tz.tzname():
+                return _JDateTimeUtils.parseTimeZone(tz.astimezone().tzname())
+
+            return _JDateTimeUtils.parseTimeZone(tz.tzname())
+        else:
+            raise Exception("Unsupported conversion: " + str(type(tz)) + " -> TimeZone")
     except Exception as e:
         raise DHError(e) from e
 
 
-def seconds_to_nanos(seconds: int) -> int:
-    """ Converts seconds to nanoseconds.
-
-    Args:
-        seconds (int): Seconds to convert.
-
-    Returns:
-        NULL_LONG if the input is NULL_LONG; otherwise the input seconds converted to nanoseconds.
-
-    Raises:
-        DHError
+def to_j_local_date(dt: Union[None, str, datetime.date, datetime.time, datetime.datetime, np.datetime64]) -> \
+        Optional[LocalDate]:
     """
-    try:
-        return _JDateTimeUtils.secondsToNanos(seconds)
-    except Exception as e:
-        raise DHError(e) from e
+    Converts a Python string, date, time, datetime, or numpy datetime64 to a Deephaven local date,
```

Comment:
Name is `to_j_local_date`, but docstring says "converts ... to a Deephaven local date". These are not consistent, and I vote for `to_dh_local_date`

Reply:
It was determined that `to_j_` was more consistent with the rest of the API.